static/app/components/feedback/summaryCategories/feedbackCategories.tsx

```diff
@@ -0,0 +1,133 @@
+   * Special case: if the label has a literal * (asterisk) in it, we have to escape it to indicate that we want to match the literal *
+   * - The asterisk escape is added after both JSON conversions, since otherwise, the \* would be escaped to \\* and would match the wildcard instead of the literal *
+   */
+
+  const exactMatchString = JSON.stringify(label);
+  let toPassToSearch = JSON.stringify(exactMatchString);
+  toPassToSearch = escapeFilterValue(toPassToSearch);
+  // Now, add the wildcards to the string (second spot and second-last spot, since we want them inside the second pair of quotes)
+  // The first and last quotes are added by the second JSON.stringify, we just manually add them back
+  toPassToSearch = `"*${toPassToSearch.slice(1, -1)}*"`;
+  return toPassToSearch;
+}
+
+function getSearchTermForLabelList(labels: string[]) {
+  labels.sort();
+  const searchTerms = labels.map(label => getSearchTermForLabel(label));
+  return `[${searchTerms.join(',')}]`;
+}
+
+export default function FeedbackCategories({
+  categories,
+}: {
+  categories: FeedbackCategory[];
+}) {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const currentQuery = useMemo(
+    () => decodeScalar(location.query.query, ''),
+    [location.query.query]
+  );
+  const searchConditions = useMemo(() => new MutableSearch(currentQuery), [currentQuery]);
+
+  const handleTagClick = (category: {
+    associatedLabels: string[];
+    primaryLabel: string;
+  }) => {
+    const allLabels = [category.primaryLabel, ...category.associatedLabels];
+
+    const exactSearchTerm = getSearchTermForLabelList(allLabels);
+
+    const isSelected = isCategoryOnlySelected(category);
+
+    const newSearchConditions = new MutableSearch(currentQuery);
+
+    if (isSelected) {
+      newSearchConditions.removeFilterValue('ai_categorization.labels', exactSearchTerm);
+    } else {
+      newSearchConditions.removeFilter('ai_categorization.labels');
+
+      // Don't escape the search term, since we want wildcards to work; escape the string ourselves before adding the wildcards
+      newSearchConditions.addFilterValue(
+        'ai_categorization.labels',
+        exactSearchTerm,
+        false
+      );
+    }
+
+    navigate({
+      ...location,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: newSearchConditions.formatString(),
+      },
+    });
+  };
+
+  const isCategoryOnlySelected = (category: {
+    associatedLabels: string[];
+    primaryLabel: string;
+  }) => {
+    // Create search terms for primary label and all associated labels
+    const allLabels = [category.primaryLabel, ...category.associatedLabels];
+    const exactSearchTerm = getSearchTermForLabelList(allLabels);
+    const currentFilters = searchConditions.getFilterValues('ai_categorization.labels');
```

Comment:
I would probably extract `ai_categorization.labels` to a constant at the top of the file so readers know it is a magic string.